styx-api-service/src/main/java/com/spotify/styx/api/BackfillResource.java

```diff
@@ -109,7 +109,7 @@ private BackfillsPayload getBackfills(RequestContext requestContext) {
 
     Stream<Backfill> backfills;
     try {
-      backfills = storage.backfills().stream();
+      backfills = storage.backfills().parallelStream();
```

Comment:
It's definitely not needed for filtering, but it looks both component and workflow are optional, so in the worst case we might end up with processing all backfills.

The other change is for sure the intensive part as we have identified.